lib/scanner/csv-scanner/csv-scanner.c

```diff
@@ -392,43 +392,40 @@ csv_scanner_is_scan_finished(CSVScanner *self)
 }
 
 void
-csv_scanner_input(CSVScanner *self, const gchar *input)
+csv_scanner_iterator_init(CSVScannerIterator *iterator, const gchar *input)
 {
-  self->src = input;
-  self->current_column = NULL;
+  iterator->src = input;
+  iterator->current_value = g_string_sized_new(128);
+  iterator->current_column = NULL;
```

Comment:
I think this is a perfect example where scratch_buffers2 would avoid another allocation in the fast path.
See pr #1368